GVFS/GVFS/CommandLine/DehydrateVerb.cs

```diff
@@ -58,6 +59,19 @@ protected override void Execute(GVFSEnlistment enlistment)
                         { nameof(this.EnlistmentRootPathParameter), this.EnlistmentRootPathParameter },
                     });
 
+                if (this.MaintenanceJob != null
+                    && this.MaintenanceJob.Equals("LooseObjects"))
+                {
+                    this.WriteMessage(tracer, "Running LooseObject Step");

```

Comment:
We have some inconsistencies with the "LooseObject[s]" string. Perhaps it should be a fixed constant?

Alternatively, we could rewrite this portion of the method as:

```
if (this.MaintenanceJob != null) {
    this.InitializeLocalCacheAndObjectsPaths(tracer, enlistment, retryConfig: null, serverGVFSConfig: null, cacheServer: null);
    PhysicalFileSystem fileSystem = new PhysicalFileSystem();
    GitRepo gitRepo = new GitRepo(
                            tracer,
                            enlistment,
                            fileSystem);
    GVFSContext context = new GVFSContext(tracer, fileSystem, gitRepo, enlistment);
    GitMaintenanceStep[] steps = new [] {
        new LooseObjectsStep(context, requireCacheLock: true);
    };
    for (int i = 0; i < steps.Length; i++) {
        if (steps[i].Area.Equals(this.MaintenanceJob)) {
            this.WriteMessage(tracer, $"Running {this.MaintenanceJob} step");
            steps[i].Execute();
        }
    }
}
```
It makes the future extension of adding the packfile step very simple.

Reply:
We definitely want the flexibility but I find the above hard to read.  I'm updating with an alternative using switch statements.

It means the MaintenanceJob my not equal the Area for the step.  I really think that's the way to go given the tradeoffs.

Reply:
I was thinking about this and if the `MaintenanceJob` parameter should be something like a comma separated list of the class names for the steps to run so that we can test an individual step?